mathesar_ui/src/utils/languageUtils.ts

```diff
@@ -110,12 +114,53 @@ export function pluralize(
   }
 }
 
+type CountWhenZero = 'numeric' | { word: string };
+type CountWhenSingular = 'hidden' | 'numeric' | { word: string };
+
+function getCountText(
+  count: number,
+  countWhenZero: CountWhenZero,
+  countWhenSingular: CountWhenSingular,
+): string {
+  if (count === 0) {
+    if (countWhenZero === 'numeric') {
+      return String(count);
+    }
+    if ('word' in countWhenZero) {
+      return countWhenZero.word;
+    }
+    assertExhaustive(countWhenZero);
+  }
+  if (count === 1) {
+    if (countWhenSingular === 'numeric') {
+      return String(count);
+    }
+    if (countWhenSingular === 'hidden') {
+      return '';
+    }
+    if ('word' in countWhenSingular) {
+      return countWhenSingular.word;
+    }
+    assertExhaustive(countWhenSingular);
+  }
+  return String(count);
+}
+
 export function labeledCount(
```

Comment:
This is the kind of function I would say should have some doc string on top of it to explain the meaning of `countWhenZero`, `countWhenSingular`, and their usages. It took me two rounds of reading to understand it completely(might be just me 😔 )

Reply:
Thanks for this feedback. I've pushed cb577407d9add146b92cdcd5b2699065656e7148 to add some more documentation to the functions in that file. Does this help?